bria-client/src/error.rs

```diff
@@ -0,0 +1,25 @@
+use thiserror::Error;
+
+#[derive(Error, Debug)]
+pub enum BriaClientError {
+    #[error("Couldn't connect to bria at url: {0}")]
+    ConnectionError(String),
+    #[error("Bria key cannot be empty")]
+    EmptyKey,
+    #[error("Couldn't create MetadataValue")]
+    CouldNotCreateMetadataValue,
+    #[error("Couldn't find address for the given external_id")]
+    AddressNotFound,
```

Comment:
When is this returned? This should be handled internally.

Reply:
when we call ```onchain_address()```, it first calls ```get_address()```. suppose we don't have an address for a given ```onchain_address_external_id``` we use ```Err(BriaClientError::AddressNotFound) => self.new_address().await``` to call ```new_address()```